pkg/gadgets/snapshot/process/types/process-collector.go

```diff
@@ -32,3 +37,100 @@ type Event struct {
 func GetColumns() *columns.Columns[Event] {
 	return columns.MustCreateColumns[Event]()
 }
+
+type nodeProcess struct {
+	event    *Event
+	children []*nodeProcess
+}
+
+func createTree(processes []*Event) (*nodeProcess, error) {
+	var root *nodeProcess
+
+	nodes := make(map[int]*nodeProcess, len(processes))
+	// Create a node for each processes.
+	for _, process := range processes {
+		nodes[process.Pid] = &nodeProcess{
+			event:    process,
+			children: make([]*nodeProcess, 0),
+		}
+	}
+
+	// Link all nodes together.
+	for _, node := range nodes {
+		ppid := node.event.ParentPid
+		if _, ok := nodes[ppid]; !ok {
+			if root != nil {
+				// Even if there are orphan process, they should have a parent process
+				// as they will get the reaper as parent process:
+				// https://elixir.bootlin.com/linux/v6.1.3/source/kernel/exit.c#L653
+				// Note that above code is both called when calling exit() system call
+				// and when receiving a fatal signal.
+				return nil, fmt.Errorf("tree has two root processes: %v and %v\nthis should not happen please verify the correctness of the tree", root, node)
+			}
+
+			root = node
+
+			continue
+		}
+
+		nodes[ppid].children = append(nodes[ppid].children, node)
+	}
+
+	if root == nil {
+		return nil, fmt.Errorf("container has no root process")
+	}
+
+	return root, nil
+}
+
+func treeToString(node *nodeProcess) string {
+	var builder strings.Builder
+	treeToStringBuilder(node, &builder, 0)
+
+	return builder.String()
+}
+
+func treeToStringBuilder(node *nodeProcess, builder *strings.Builder, depth int) {
+	fmt.Fprintf(builder, "%s|-%s(%d)\n", strings.Repeat("\t", depth), node.event.Command, node.event.Pid)
+	if len(node.children) == 0 {
+		return
+	}
```

Comment:
This is not needed. The for loop on next line will not iterate if the slice is empty.